src/Play.ts

```diff
@@ -0,0 +1,573 @@
+import { Bonus, Game, Letter, PlayerIndex, SparseArray } from "./types";
+import words from "./words.json";
+import {
+  canReachPlayer,
+  horizontalSlice,
+  isValidMatch,
+  sliceToRegex,
+  verticalSlice,
+} from "./utils/gameBoardHelper";
+import { tileValues } from "./constants";
+
+let lastAwaitTime = 0;
```

Comment:
idk how to do singletons in javascript so if have a better idea, go for it.

If I understand javascript's event loop correctly, then this is here so that any task that needs to yield to event loop doesn't do it so often that it takes more processing time to manage the loop than to actually compute our plays. I added this and computing an already computed play turned from few seconds to basically instant.